File: 796. Rotate String/796. Rotate String.js
/**
 * @param {string} A
 * @param {string} B
 * @return {boolean}
 */
var rotateString = function(A, B) {
    if (A.length !== B.length) return false;
    
    var size = A.length;
    
    if(size === 0) return true;
    
    for(var i = 0; i < A.length; i++) {
        var result = true
        
        for(var j = 0; j < A.length; j++) {
            if(A[j] !== B[(j + i) % size]) {
                result = false;
                break;
            }
        }
        
        if(result) return true;
    }
    
    return false;
};

File: 796. Rotate String/796. Rotate String.rb
# @param {String} a
# @param {String} b
# @return {Boolean}
def rotate_string(a, b)
	return false if a.size != b.size
	a, b = a.split(""), b.split("")
	result = true
	(a.size + 1).times do |i|
		result = true
		a.each_with_index do |val, index|
			if val != b[(i + index) % a.size]
				result = false
				break
			end
		end
		if result
			return result
		end
	end
	result
end
